test/test-loader.js

```diff
@@ -456,4 +456,22 @@ exports['test loader on unsupported modules with checkCompatibility default'] =
   unload(loader);
 };
 
+exports['test lazy globals'] = function (assert) {
+  let uri = root + '/fixtures/loader/lazy/';
+  let gotFoo = false;
+  let foo = {};
+  let globals = {
+    get foo() {
+      gotFoo= true;
+      return foo;
+    }
+  };
+  let loader = Loader({ paths: { '': uri }, globals: globals});
+  assert.ok(!gotFoo, "foo hasn't been accessed during loader instanciation");
+  let program = main(loader, 'main');
+  assert.ok(!gotFoo, "foo hasn't been accessed during module loading");
+  assert.equal(program.useFoo(), foo, "foo mock works");
+  assert.ok(gotFoo, "foo has been accessed only when we first try to use it");
+};
```

Comment:
I think you may have mixed something up here. Your changes make access to built-in modules lazy that is passed as `modules` option to a loader. While here you test that `globals` are lazy instead. I imagine that's also why the tests are failing on travis.